grapher/text/MarkdownTextWrap.tsx

```diff
@@ -0,0 +1,527 @@
+        }
+    })
+}
+
+interface MarkdownTextWrapHTMLProps {
+    isSVG?: false
+    x?: number
+    y?: number
+    svgOptions?: React.SVGProps<SVGTextElement>
+}
+
+interface MarkdownTextWrapSVGProps {
+    isSVG: true
+    x: number
+    y: number
+    svgOptions?: React.SVGProps<SVGTextElement>
+}
+
+type MarkdownTextWrapProps = {
+    text: string
+    maxWidth: number
+    lineHeight?: number
+    fontSize: number
+    fontWeight?: number
+} & (MarkdownTextWrapHTMLProps | MarkdownTextWrapSVGProps)
+
+export class MarkdownTextWrap extends React.Component {
+    props: MarkdownTextWrapProps
+    constructor(props: MarkdownTextWrapProps) {
+        super(props)
+        this.props = props
+    }
+
+    @computed get maxWidth(): number {
+        return this.props.maxWidth ?? Infinity
+    }
+    @computed get lineHeight(): number {
+        return this.props.lineHeight ?? 1.1
+    }
+    @computed get fontSize(): number {
+        return this.props.fontSize ?? 1
+    }
+    @computed get fontWeight(): number | undefined {
+        return this.props.fontWeight
+    }
+    @computed get text(): string {
+        return this.props.text
+    }
+    @computed get ast(): MarkdownRoot["children"] {
+        const result = mdParser.markdown.parse(this.props.text)
+        if (result.status) {
+            return result.value.children
+        }
+        return []
+    }
+
+    @computed get lines(): IRToken[][] {
+        const tokens = parsimmonToTextTokens(this.ast)
+        return splitIntoLines(tokens, this.maxWidth)
+    }
+
+    // @computed get height(): number {
+    //     if (this.lines.length === 0) return 0
+
+    //     return (
+    //         this.lines.reduce((total, line) => total + line.height, 0) +
+    //         this.lineHeight * (this.lines.length - 1)
+    //     )
+    // }
+
+    // @computed get width(): number {
+    //     return max(this.lines.map((l) => l.width)) ?? 0
+    // }
+
+    @computed get htmlStyle(): any {
+        const { fontSize, fontWeight, lineHeight } = this
+        return {
+            fontSize: fontSize.toFixed(2) + "px",
+            fontWeight: fontWeight,
+            lineHeight: lineHeight,
+            overflowY: "visible",
+        }
+    }
+
+    render(): JSX.Element | null {
+        const { props, lines, fontSize, fontWeight, lineHeight } = this
+
+        // if (lines.length === 0) return null
+
+        if (props.isSVG) {
+            const yOffset = 1
+            // const yOffset = props.y + lines[0].height - lines[0].height * 0.2
+            return (
+                <text
+                    fontSize={fontSize.toFixed(2)}
+                    fontWeight={fontWeight}
+                    x={props.x.toFixed(1)}
+                    y={yOffset.toFixed(1)}
+                    {...props.svgOptions}
+                >
+                    {lines.map((line) =>
+                        line.map((token, i) => token.toSVG(i))
+                    )}
```

Comment:
We have to wrap each line in a `<tspan>` like `TextWrap` does it. 

We could either do that here directly, or we could also create an `IRLine` token that wraps in `<tspan>`, but it would need to receive the offsets as well... probably option 1 is better.